### PR TITLE
fix(rules): Fix user in AlertRuleSerializer

### PR DESCRIPTION
**Pull request by ceorourke**
`user` was sometimes being passed to the `AlertRuleSerializer` as a `User` or `RpcUser` object, and sometimes it was a dictionary. This caused [this issue](https://sentry.sentry.io/issues/4457973147/) which led to an incident since the alert rule details page wouldn't load. This was "fixed" [here](https://github.com/getsentry/sentry/commit/2936554e63c58f2f0f898d4eb320fc5d70ce0ca4) by handling both types, but that wasn't ideal since we shouldn't be passing around different variable types for one variable in the first place.

Digging deeper I found that `user` was reassigned [here](https://github.com/getsentry/sentry/blob/ede26403b8a2dc127867d2c68a5b30a57b05535d/src/sentry/api/serializers/models/alert_rule.py#L90) if an `rpc_user` was found, so in [this PR](https://github.com/getsentry/sentry/pull/55868/files) that added the `RuleSnooze` data to the response, when `latestIncident` was passed to `expand` the `user` that was being passed to the serializer again was a dictionary rather than the expected `User` or `RpcUser`, so the query wouldn't work. 

This PR fixes all of that by not re-assigning `user` - when it was originally done we didn't use `user` in a meaningful way later on in the code so it didn't matter. 

I also changed the variable name `type` since as I was debugging this, I couldn't print the type of `user` which was really annoying. (e.g. `print(type(user))` threw an error that `type` hadn't been defined yet).

### CODE REVIEW

File: src/sentry/api/serializers/models/alert_rule.py [hunk 1]
```diff
@@ -83,10 +82,12 @@ def get_attrs(self, item_list, user, **kwargs):
         for rule_activity in rule_activities:
             rpc_user = user_by_user_id.get(rule_activity.user_id)
```
Comment (JoshFerge) [inline src/sentry/api/serializers/models/alert_rule.py:83]: why do we have to use the rpc_user at all here? are alert rules in control or region silos?

**Review by JoshFerge — APPROVED** [review]
assuming we have tests that cover this?

File: src/sentry/api/serializers/models/alert_rule.py (hunk 1 above)
Comment (ceorourke) [reply src/sentry/api/serializers/models/alert_rule.py:83]: alert rules are in the region silo

Comment (ceorourke) [reply src/sentry/api/serializers/models/alert_rule.py:83]: also hc was the one who added this stuff

Comment (JoshFerge) [reply src/sentry/api/serializers/models/alert_rule.py:83]: oh interesting.

File: tests/sentry/incidents/endpoints/test_project_alert_rule_index.py [hunk 2]
```diff
@@ -156,6 +156,26 @@ def test_project_not_in_request(self):
 class ProjectCombinedRuleIndexEndpointTest(BaseAlertRuleSerializerTest, APITestCase):
     endpoint = "sentry-api-0-project-combined-rules"
 
+    def test_expand_latest_incident(self):
```
[the remark this answers is not in the document]
Comment (ceorourke) [reply tests/sentry/incidents/endpoints/test_project_alert_rule_index.py:159]: [This test](https://github.com/getsentry/sentry/blob/master/tests/sentry/incidents/endpoints/test_organization_combined_rule_index_endpoint.py#L785) also covers that path

File: src/sentry/api/serializers/models/alert_rule.py [hunk 3]
```diff
@@ -241,7 +234,13 @@ def get_attrs(self, item_list, user, **kwargs):
             if isinstance(item, AlertRule):
                 alert_rule = serialized_alert_rules.pop(0)
                 if "latestIncident" in self.expand:
-                    alert_rule["latestIncident"] = incident_map.get(item.incident_id)
+                    incident = (
```
Comment (malwilley) [inline src/sentry/api/serializers/models/alert_rule.py:237]: Was this change necessary? Seems unrelated to the user changes? 

It also seems a bit strange to make both this query and the one above that fills `incident_map`, but LMK if I'm missing something (I probably am)

Comment (ceorourke) [reply src/sentry/api/serializers/models/alert_rule.py:237]: I could do it in 2 separate PRs if that makes more sense, but this one is a subclass of `AlertRuleSerializer` and since I noticed the `user` issue only when `latestIncident` was passed to `expand`, I wanted to clean up other instances of it. I had to rewrite the queries because there is no `incident_id` column on an `AlertRule` model.

Comment (malwilley) [reply src/sentry/api/serializers/models/alert_rule.py:237]: Makes sense to me 👍

**Review by malwilley — APPROVED** [review]
Good cleanup 👏
